lib/pacemaker/pcmk_sched_instances.c

```diff
@@ -679,18 +678,19 @@ find_compatible_instance_on_node(const pe_resource_t *match_rsc,
     for (GList *iter = instances; iter != NULL; iter = iter->next) {
         pe_resource_t *instance = (pe_resource_t *) iter->data;
 
-        if (pcmk__instance_is_compatible(instance, node, role, current)) {
-            pe_rsc_trace(match_rsc,
-                         "Found %s instance %s compatible with %s on %s",
+        if (pcmk__instance_matches(instance, node, role, current)) {
+            pe_rsc_trace(match_rsc, "Found %s %s instance %s compatible with %s on %s",
```

Comment:
overlong now (87) after moving it up a line since the previous version; your call